streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -896,26 +904,23 @@ private void connectStateStoreNameToSourceTopicsOrPattern(final String stateStor
             }
         }
 
-
-        if (sourceTopics.isEmpty() && sourcePatterns.isEmpty()) {
-            throw new TopologyBuilderException("can't find source topic or source Pattern for state store " +
-                    stateStoreName);
+        //If both aren't empty it will force and update to the map which according to the warning above we should not do.
+        if (!sourceTopics.isEmpty() && !sourcePatterns.isEmpty()) {
```

Comment:
I'm not very sure I understand your previous comment about preventing this: suppose we have the following topology:

```
sourceNode1: pattern "topicA*"
sourceNode2: topics ["topicBB"]
processNode3: parents ["sourceNode1", "sourceNode2"]
```

And we are adding a `stateStore` to `processNode3`, we will trace back to both sourceNode1 and sourceNode2, then after the first fetch when metadata is updated as `topicAA, topicAC, topicBB`, this state store should be linked to all these topics. Right?